lib/fog/vcloud_director/generators/compute/compose_common.rb

```diff
@@ -115,6 +122,68 @@ def build_source_items(xml)
             xml.AllEULAsAccepted (@configuration[:AllEULAsAccepted] || true)
           end
 
+          def build_virtual_hardware_section(xml, hardware)
+            xml[:ovf].VirtualHardwareSection do
+              xml[:ovf].Info('Virtual hardware requirements')
+              virtual_hardware_section_item_mem(xml, **hardware[:memory]) if hardware[:memory]
+              virtual_hardware_section_item_cpu(xml, **hardware[:cpu]) if hardware[:cpu]
+              if hardware[:disk]
```

Comment:
Do you think one could set the disk as a single object instead of an array with a single object? Would it make sense to check this here and make sure the object is also treated properly?

Reply:
Alternatively, I think we should use the name `:disks` to indicate there are more of them.

Reply:
I like both your suggestions, I'll rename to disks and support either list or dict directly.